static/graphs/obsidian.js

```diff
@@ -0,0 +1,277 @@
+    d3
+      .forceLink(linksData)
+      .id((d) => d.id)
+      .distance(70)
+  )
+  .force("center", d3.forceCenter(width / 2, height / 2))
+  .force("collision", d3.forceCollide().radius(80))
+  .stop();
+
+const g = svg.append("g");
+let link = g.append("g").attr("class", "links").selectAll(".link");
+let node = g.append("g").attr("class", "nodes").selectAll(".node");
+let text = g.append("g").attr("class", "text").selectAll(".text");
+
+const resize = () => {
+  if (d3.event) {
+    const scale = d3.event.transform;
+    zoomLevel = scale.k;
+    g.attr("transform", scale);
+  }
+
+  const zoomOrKeep = (value) => (zoomLevel >= 1 ? value / zoomLevel : value);
+
+  const font = Math.max(Math.round(zoomOrKeep(FONT_SIZE)), 1);
+
+  text.attr("font-size", `${font}px`);
+  text.attr("y", (d) => d.y - zoomOrKeep(FONT_BASELINE + nodeSize[d.id]));
+  link.attr("stroke-width", zoomOrKeep(STROKE));
+  node.attr("r", (d) => {
+    return zoomOrKeep(nodeSize[d.id])
+  })  
+  svg
+    .selectAll("circle")
+    .filter((_d, i, nodes) => d3.select(nodes[i]).attr("active"))
+    .attr("r", (d) => zoomOrKeep(ACTIVE_RADIUS_FACTOR * nodeSize[d.id]));
+
+  document.getElementById("zoom").innerHTML = zoomLevel.toFixed(2);
+};
+
+window.addEventListener("message", (event) => {
+  const message = event.data;
+
+  switch (message.type) {
+    case "refresh":
+      const { nodes, edges } = message.payload;
+
+      if (sameNodes(nodesData, nodes) && sameEdges(linksData, edges)) {
+        return;
+      }
+
+      nodesData = nodes;
+      linksData = edges;
+      restart();
+      break;
+    case "fileOpen":
+      let path = message.payload.path;
+      if (path.endsWith(".git")) {
+        path = path.slice(0, -4);
+      }
+
+      const fixSlashes = (input) => {
+        const onLocalWindowsFilesystem =
+          navigator.platform == "Win32" && /^\w:\\/.test(input);
+        return onLocalWindowsFilesystem ? input.replace(/\//g, "\\") : input;
+      };
+
+      node.attr("active", (d) => (fixSlashes(d.path) === path ? true : null));
+      text.attr("active", (d) => (fixSlashes(d.path) === path ? true : null));
+      break;
+  }
+
+  // Resize to update size of active node
+  resize();
+});
+
+const ticked = () => {
+  document.getElementById("connections").innerHTML = linksData.length;
+  document.getElementById("files").innerHTML = nodesData.length;
+
+  node.attr("cx", (d) => d.x).attr("cy", (d) => d.y);
+  text.attr("x", (d) => d.x).attr("y", (d) => d.y - (FONT_BASELINE - nodeSize[d.id]) / zoomLevel);
+  link
+    .attr("x1", (d) => d.source.x)
+    .attr("y1", (d) => d.source.y)
+    .attr("x2", (d) => d.target.x)
+    .attr("y2", (d) => d.target.y);
+};
+
+const restart = () => {
+  updateNodeSize()
+  node = node.data(nodesData, (d) => d.id);
+  node.exit().remove();
+  node = node
+    .enter()
+    .append("circle")
+    .attr("r", (d) => {
+      return nodeSize[d.id]
+    })
+    .on("click", onClick)
+    .on("mouseover", mouseover)
```

Comment:
Another very minor thing, but I think we could use some more consistent naming here.

`click` gives us `onClick` while `mouseover` leads to `mouseover`. Personally I would go with `on[...]` plus camelCase for all, like `onClick`, `onMouseOver` etc. but it is just a matter of my personal preference so feel free to go with whatever consistent way you prefer.